src/deprecated.jl

```diff
@@ -1285,3 +1285,18 @@ import Base: |>
 @deprecate colwise(f) x -> colwise(f, x)
 @deprecate groupby(cols::Vector{T}; sort::Bool = false, skipmissing::Bool = false) where {T} x -> groupby(x, cols, sort = sort, skipmissing = skipmissing)
 @deprecate groupby(cols; sort::Bool = false, skipmissing::Bool = false) x -> groupby(x, cols, sort = sort, skipmissing = skipmissing)
+
+function Base.getindex(x::AbstractIndex, idx::Bool)
+    Base.depwarn("Indexing with Bool values is deprecated except for Vector{Bool}", :getindex)
+    1
+end
+
+function Base.getindex(x::AbstractIndex, idx::Real)
+    Base.depwarn("Indexing with values that are not Integer is deprecated", :getindex)
+    Int(idx)
+end
+
+function Base.getindex(x::AbstractIndex, idx::AbstractVector{Union{Bool, Missing}})
```

Comment:
Woops, actually I think we should allow `AbstractVector{Union{Bool, Missing}` but print a deprecation warning for missing values. That's kind of a special case which hasn't been considered in Base yet, but probably should for convenience.

Reply:
What do you expect the functionality in Base will be then (so that I know what to implement).

But again - in general I think is better not to handle it at all but delegate to Base and treat all this code as transitory aiming in long run to write in the documentation something like:
> column indexing works like all indexing of vectors in Base except that also indexing by `Symbol` or `Vector{Symbol}` is allowed

I would even go further and maybe even target to say:
> column indexing works like all indexing of `NamedTuple` in Base

which can be done even if internally we do not switch to `NamedTuple` (but if we do then it will be given out-of-the-box). The difference is that `NamedTuple` does not support indexing by `Vector{Byte}` and `Vectors` do.

Reply:
> What do you expect the functionality in Base will be then (so that I know what to implement).

As I noted for now I think we should print a deprecation warning when missing values are found, replacing them with `false` as we do now to avoid breaking code. Then we could make them an error in the next major release.

It would be nice to be fully consistent with Base, but I'm not sure the indexing behavior of `NamedTuple` is completely settled yet. If we find inconsistencies we might want to discuss whether to change Base if we find it useful.

Reply:
fixed